pkg/api/store/workload/transform_store.go

```diff
@@ -38,9 +38,9 @@ func NewTransformStore(store types.Store) types.Store {
 			nodeName := convert.ToString(values.GetValueN(data, "nodeId"))
 			if nodeName != "" {
 				state := getState(data)
-				_, nodeName := ref.Parse(state[getKey(nodeName)])
-				data["nodeId"] = nodeName
-				values.PutValue(data, nodeName, "scheduling", "node", "nodeId")
+				_, nodeId := ref.Parse(state[getKey(nodeName)])
+				data["nodeId"] = nodeId
+				values.PutValue(data, state[getKey(nodeName)], "scheduling", "node", "nodeId")
```

Comment:
why do we read from the state once again here? Shouldn't the value be the same as nodeId?